Classes/FieldProcessor/PageUidToHierarchy.php

```diff
@@ -68,10 +68,14 @@ public function process(array $values)
         $results = [];
 
         foreach ($values as $value) {
-            list($rootPageUid, $mountPoint) = GeneralUtility::trimExplode(',',
-                $value, true, 2);
-            $results[] = $this->getSolrRootlineForPageId($rootPageUid,
-                $mountPoint);
+            $results[] = $this->getSolrRootlineForPageId(
+                ...GeneralUtility::trimExplode(
+                    ',',
+                    $value,
+                    true,
+                    2
+                )
+            );
```

Comment:
Is that change really wanted?
See: https://scrutinizer-ci.com/g/TYPO3-Solr/ext-solr/inspections/5efe43ac-cf9b-4387-90e7-704f8507fb30/issues/files/Classes/FieldProcessor/PageUidToHierarchy.php?status=new&selectedLabels%5B0%5D=9&orderField=path&order=asc&fileId=Classes/FieldProcessor/PageUidToHierarchy.php&honorSelectedPaths=0

I would like to avoid the new scrutinizer issues if possible.

Reply:
GeneralUtility::trimExplode creates an array with one or two keys.  
The method getSolrRootlineForPageId expects one or two arguments. 
The `...` can be used when calling functions to unpack an array or Traversable variable or literal into the argument list.  
See [php manual](https://www.php.net/manual/en/functions.arguments.php#example-162).

The old code produces errors when trimExplode returns an array with only one key.
```php
list($rootPageUid, $mountPoint) = GeneralUtility::trimExplode(',', $value, true, 2);
$results[] = $this->getSolrRootlineForPageId($rootPageUid, $mountPoint);
```

When the `...` argument list is a problem, i can change the code to this:
```php
$args = GeneralUtility::trimExplode(',', $value, true, 2);
$results[] = $this->getSolrRootlineForPageId($args[0] ?? 0, $args[1] ?? '');
```

Let me know how we want to proceed.